### PR TITLE
[Merged by Bors] - Instrument misc

### PR DESCRIPTION
**Pull request by nicholastmosher**
- Adds `#[tracing::instrument]` to more areas in misc crates to support better tracing e.g. with otel/jaeger

### CODE REVIEW

**Comment by sehz** [comment]
rebase to get clean diff

**Comment by nicholastmosher** [comment]
@sehz rebased

File: src/protocol/fluvio-protocol-api/src/request.rs [hunk 1]
```diff
@@ -132,6 +135,7 @@ impl<R> Decoder for RequestMessage<R>
 where
     R: Request,
 {
+    #[instrument(level = "trace", skip(self, src, version))]
     fn decode<T>(&mut self, src: &mut T, version: Version) -> Result<(), IoError>
```
Comment (sehz) [inline src/protocol/fluvio-protocol-api/src/request.rs:139]: These are very low level call, it just add lots of noise, is there any benefit to add tracing?

**Review by sehz — CHANGES_REQUESTED** [review]
minor nits

**Comment by nicholastmosher** [comment]
I took a look at Jaeger and didn't see any place where these spans were really showing up, so for now they're not adding any value, I'll remove them.

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**